### PR TITLE
Add error code flags

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @brson 

cc @steveklabnik 
cc @jonathandturner

### CODE REVIEW

File: src/librustc/diagnostics.rs [hunk 1]
```diff
@@ -947,6 +945,8 @@ position that needs that trait. For example, when the following code is
 compiled:
 
 ```compile_fail
+#![feature(on_unimplemented)]
+
```
Comment (brson) [inline src/librustc/diagnostics.rs]: Why did these examples start requiring `#![feature(on_unimplemented)]` but still not  get error code annotations?

File: src/librustc/diagnostics.rs [hunk 2]
```diff
@@ -470,11 +469,13 @@ parameters are involved, this cannot always be done.
 
 So, for example, the following is not allowed:
 
-```compile_fail
+```
```
Comment (brson) [inline src/librustc/diagnostics.rs]: Why did this case lose the compile_fail annotation?

Comment (GuillaumeGomez) [inline src/librustc/diagnostics.rs]: Like said [above](https://github.com/rust-lang/rust/pull/34401/files#diff-48fe837d9fa7d777a951c5853bb98474R463), this error doesn't exist anymore. I didn't remove it in case people would still need it for whatever reason.

File: src/librustc/diagnostics.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/librustc/diagnostics.rs]: Because to use `#[rustc_on_unimplemented]`, you need to enable the feature. However, the code throws `E0230`, so I can't add `E0272`. The same goes for `E0273` and `E0274`.

File: src/librustc/diagnostics.rs [hunk 3]
```diff
@@ -675,12 +676,9 @@ E0269: r##"
...
 
-```compile_fail
-fn foo(x: u8) -> u8 {
-    if x > 0 {
-        x // alternatively, `return x`
-    }
-    // nothing here
+```compile_fail,E0269
+fn abracada_FAIL() -> String {
+    "this won't work".to_string();
```
Comment (kumbayo) [inline src/librustc/diagnostics.rs]: Is this change intentional?
It seems the source code does not match the following explanation text anymore.

Comment (GuillaumeGomez) [inline src/librustc/diagnostics.rs]: Ah damn, i didn't pay attention to the explanation. Want to fix it?

Comment (kumbayo) [inline src/librustc/diagnostics.rs]: Sorry, it don't have enough time/knowledge to fix this in short time frame.

Comment (GuillaumeGomez) [inline src/librustc/diagnostics.rs]: I'll then. Thanks for notifying me!
